### PR TITLE
Filtering championships by city

### PR DESCRIPTION
**Pull request by Mickss**

### CODE REVIEW

File: src/main/java/org/micks/champmaker/championships/ChampionshipController.java [hunk 1]
```diff
@@ -38,8 +42,9 @@ public void registerTeamToChampionship(@PathVariable long champId, @RequestBody
     }
 
     @GetMapping
-    public List<ChampionshipDTO> getChampionships() {
-        return championshipService.getChampionships();
+    public List<ChampionshipDTO> getChampionships(@RequestParam(required = false) String city) {
+        log.info("Fetching city for params: {}", city);
```
Comment (tknowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipController.java]: Should log `fetching championships`, not `city`.

Comment (Mickss) [reply src/main/java/org/micks/champmaker/championships/ChampionshipController.java]: fixed

Comment (kbnowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipController.java]:
```suggestion
        log.info("Fetching championships for params: {}", city);
```

File: src/main/java/org/micks/champmaker/championships/ChampionshipService.java [hunk 2]
```diff
@@ -35,9 +35,10 @@ public ChampionshipDTO getChampionship(long champId) throws EntityNotFoundExcept
         }
     }
 
-    public List<ChampionshipDTO> getChampionships() {
+    public List<ChampionshipDTO> getChampionships(String city) {
         List<ChampionshipEntity> championshipList = championshipRepository.findAll();
         return championshipList.stream()
+                .filter(championshipEntity -> city == null || championshipEntity.getCity().contains(city))
```
Comment (kbnowak) [inline src/main/java/org/micks/champmaker/championships/ChampionshipService.java:41]: why contains and not equals?

File: src/main/java/org/micks/champmaker/championships/ChampionshipController.java (hunk 1 above)
Comment (Mickss) [reply src/main/java/org/micks/champmaker/championships/ChampionshipController.java]: fixed

**Review by tknowak — APPROVED** [review]

**Review by kbnowak — APPROVED** [review]

**Pull request merged**